File: src/main/java/org/ariane/thirdlab/dao/PostTagRelationDao.java
package org.ariane.thirdlab.dao;

import org.ariane.thirdlab.domain.PostTagRelation;

import java.util.List;

public interface PostTagRelationDao extends AbstractDao<PostTagRelation> {

	public List<PostTagRelation> findRelationsByTagId(long tagId);

	public List<PostTagRelation> findRelationsByPostId(long postId);

}

File: tl/dao/postTagRelationDao.go
package dao

import (
	"github.com/jinzhu/gorm"
	"thirdlab/tl/domain"
)

func FindPostTagRelationByPostId(db *gorm.DB, postId int) []*domain.PostTagRelation {
	relations := make([]*domain.PostTagRelation, 0)
	db.Where(&domain.PostTagRelation{PostId: postId}).
		Find(&relations)
	return relations
}

func FindPostTagRelationByTagId(db *gorm.DB, tagId int) []*domain.PostTagRelation {
	relations := make([]*domain.PostTagRelation, 0)
	db.Where(&domain.PostTagRelation{TagId: tagId}).
		Find(&relations)
	return relations
}